### PR TITLE
Consumable Treasure Stack Size - DO NOT MERGE

### PR DESCRIPTION
**Pull request by mhn65536**
Allows you to specify that treasurechests and npcs with consumables giv you mor than 1 item.

### CODE REVIEW

File: FF1Blazorizer/Tabs/TreasuresTab.razor [hunk 1]
```diff
@@ -1,55 +1,58 @@
...
+		<div class="checkbox-cell"></div>
+		<TriStateCheckBox UpdateToolTip="@UpdateToolTipID" Id="FreeLute" IsEnabled="@(!Flags.ShortToFR)" @bind-Value="Flags.FreeLuteFlag">Free Lute</TriStateCheckBox>
+		<TriStateCheckBox UpdateToolTip="@UpdateToolTipID" Id="FreeTail" IsEnabled="@(Flags.NoTail != true)" @bind-Value="Flags.FreeTail">Free Tail</TriStateCheckBox>
+		<CheckBox UpdateToolTip="@UpdateToolTipID" Id="freeOrbsCheckBox" IsEnabled="@Flags.FreeOrbsEnabled" @bind-Value="Flags.FreeOrbs">Free Orbs</CheckBox>
+		<div class="checkbox-cell"></div>
+		<h4>Bans</h4>
+		<TriStateCheckBox UpdateToolTip="@UpdateToolTipID" IsEnabled="@(!Flags.FreeTail)" Id="NoTailCheckBox" @bind-Value="Flags.NoTail">Remove Tail</TriStateCheckBox>
+		<TriStateCheckBox UpdateToolTip="@UpdateToolTipID" Id="NoMasamuneCheckBox" @bind-Value="Flags.NoMasamune">Remove Masamune</TriStateCheckBox>
+		<div class="checkbox-cell"></div>
+		<h4>Consumables</h4>
```
Comment (wildham0) [inline FF1Blazorizer/Tabs/TreasuresTab.razor:44]: Should probably be put with Starting Inventory

File: FF1Lib/Hacks.cs [hunk 2]
```diff
@@ -1044,7 +1044,7 @@ public void MonsterInABox(MT19337 rng, Flags flags)
 			PutInBank(0x1F, 0xDD78, Blob.FromHex("A9002003FEA645BD00B18561A9112003FE20B08E8A60"));
 
 			// Check for trapped monster routine, see 11_8EC0_CheckTrap.asm
-			PutInBank(0x11, 0x8EB0, Blob.FromHex("A561202096B030A645BD008FF025856AA9C0203D96A56A200096A903CD866BD00820189668684C43961820E98E201896A2F06020E98E60AA60A911855818A5612093DDA445B90062090499006260"));
+			PutInBank(0x11, 0x8EB0, Blob.FromHex("A561202096B02DA645BD008FF022856AA9C0203D96A56A200096A903CD866BD0062018964C439620E68E201896A2F06020E68E60AA60A911855818A5612093DDA445B90062090499006260"));
```
Comment (wildham0) [inline FF1Lib/Hacks.cs:1047]: What are the changes? I know this also had some bugfixes recently

File: FF1Lib/Treasure.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ public partial class FF1Rom : NesRom
 		public const int TreasureCount = 256;
 
 		public const int lut_MapObjTalkJumpTblAddress = 0x390D3;
-		public const string giveRewardRoutineAddress = "93DD";
+		public const string giveRewardRoutineAddress = "06B0";
```
Comment (wildham0) [inline FF1Lib/Treasure.cs:17]: so if we do that we need to update almost all the talkroutines, which will be a pain, I'd suggest putting a jump at DD93 to B006 instead

File: FF1Lib/asm/11_B006_GiveReward.asm [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  @Item:                       ; (5 + 15 + 7 bytes)
+    TAX                        ; put item ID in X AA
+    CMP #$0C                   ; then check for canal C90C
+    BNE :+                     ; If canal then D005
+      DEC unsram, X            ; decrement DE0060
+      BCS @WasCanal            ; and open it B018
+:   CMP #$36                   ; CMP ItemId with Tent C936
+    BCC :+                     ; < Tent normal Add One routine 9011
+		SBC #$36               ; Subtract TentOffset E936
+		TAY                    ; Punt Index into Y A8
```
Comment (wildham0) [inline FF1Lib/asm/11_B006_GiveReward.asm:23]: sometime the game expect registers to be left alone, do we need to backup Y's value here?

File: FF1Lib/asm/1F_DD78_OpenTreasureChestRewrite.asm [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
 
     LDX tileprop+1           ; put chest index in X A645
-    LDA lut_Treasure, X      ; use it to get the contents of the chest BD00B1
+    LDY lut_Treasure, X      ; use it to get the contents of the chest BC00B1
 
-    JSR GiveReward            ; Jump to new sub routine 2094DD
+	LDA #BANK_TALKROUTINE    ; swap to bank containing new talkroutines A911
+    JSR SwapPRG_L            ; 2003FE
+	
+	TYA						; Put ItemId into A 98
```
Comment (wildham0) [inline FF1Lib/asm/1F_DD78_OpenTreasureChestRewrite.asm:11]: looks like not

**Review by wildham0 — CHANGES_REQUESTED** [review]
we should do an alpha with this first, then is should be good

File: FF1Lib/Hacks.cs (hunk 2 above)
Comment (mhn65536) [reply FF1Lib/Hacks.cs:1047]: There's actually a merge error here. I replaced 93DD and lost it while conflict resolving.

File: FF1Lib/Treasure.cs (hunk 3 above)
Comment (mhn65536) [reply FF1Lib/Treasure.cs:17]: Should not be a problem, as neither DD93 nor 93DD exists in the randomizer files, with the above exception, that i will corect.

Comment (mhn65536) [reply FF1Lib/Treasure.cs:17]: DD93 most likely is compromised, as i inserted some instructions in the OpenTreasureChest routine

File: FF1Blazorizer/Tabs/TreasuresTab.razor (hunk 1 above)
Comment (mhn65536) [reply FF1Blazorizer/Tabs/TreasuresTab.razor]: ok

File: FF1Lib/Hacks.cs (hunk 2 above)
Comment (mhn65536) [reply FF1Lib/Hacks.cs]: It now should be the correct blob, with the changed address. Please check the blob, with your version just to make sure.

**Comment by mhn65536** [comment]
Since it's still a cross rep PR, can you deploy it for an alpha test, once it's ready.
@wildham0

**Pull request closed without merging**